### DISCUSSION TITLE
Current font does not support math on Noto Sans Math

### DISCUSSION THREAD

**Discussion by soupslurpr** [typst/typst#2612] [General]
### Description

I'm including Noto Sans Math in the fonts and I set it using #set text(font: "Noto Sans Math"). I see the change to the font, but when I try to use math it gives me the `current font does not support math`. Can anyone help please?

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
You're likely on an old version of noto sans math. It didn't use to have an opentype math table.

That being said, I'd avoid using it even if you could get it working. It has a lot of issues.

Edit:
Do you mean `show math.equation: set text(...)`? Simply setting the text font shouldn't affect math.

**Comment by soupslurpr** [comment]
I mean something like $ 123 $. I'm using the latest from Google fonts, so that's weird. Are there any math fonts you can recommend? I need them to have a permissive license such as open font license. Thanks

**Comment by Enivex** [comment]
> I mean something like $ 123 $. I'm using the latest from Google fonts, so that's weird. Are there any math fonts you can recommend? I need them to have a permissive license such as open font license. Thanks

You may have to get a development version from GitHub to get the math table.

The default math font (new computer modern math) is decent-ish. Personally I prefer stix two math. If you'd like a sans serif font there's fira math.

**Reply by soupslurpr** [reply]
I tried the latest release from the GitHub and it still didn't work. I tried STIX Two Math and I got the same error

**Reply by Enivex** [reply]
Using `#show math.equation: set text(font: "Stix Two Math")` ?

**Reply by soupslurpr** [reply]
Ohhh that did the trick! Thank you so much... I was setting it normally lol

**Comment by alerque** [comment]
Google Fonts is *sometimes* the latest version of a font (especially if Google funded fixing up a font and only published the cleanups themselves) but it is quite frequently also *not* the latest upstream version. In a few cases the versions there are downright archaic. There are several factors making that the case, but one is that they basically don't update fonts there after first publishing if the changes would update metrics. Meanwhile upstream font projects sometimes move on and keep developing.

If you like a font, it's often worth looking up what the upstream project is and if they have releases outside of Google Fonts.

Also you might check out Libertinus Math for something style wise similar to Typst's default font (since it's literally a continuation of the same font project).

**Reply by soupslurpr** [reply]
I couldn't get Libertinus Math to work either (downloaded from https://github.com/alerque/libertinus/releases/tag/v7.040) maybe I am doing something wrong in my font loading code?

**Reply by alerque** [reply]
Yes, it sounds like you ar. Where did you put the font files?

```typst
#show math.equation: set text(font: "Libertinus Math")
```

...works just fine for me.

**Reply by soupslurpr** [reply]
I solved it, I was using normal set text and not the math one 😅 thanks for the help though

**Comment by pacaunt** [comment]
Can anyone provide the link to that development version? 
Currently I installed v.3 of Noto Sans Math font from ![this github](https://github.com/notofonts/math/tree/main) but still cannot make it work.
